kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumingService.java

```diff
@@ -116,4 +130,36 @@ private boolean filterTopicMessage(TopicMessageEvent message, String query) {
         || (!StringUtils.isEmpty(msg.getContent()) && msg.getContent().contains(query));
   }
 
+  private void evalJsFilterFn(String jsFilterFn) {
+    try {
+      jsEngine.eval(jsFilterFn);
+    } catch (ScriptException e) {
+      throw new UnprocessableEntityException(getMessageWithJsFilterFnTip(e.getMessage()));
+    }
+  }
+
+  private boolean jsFilterTopicMessageEvent(InternalTopicMessageEvent messageEvent,
+                                            String jsFilterFn) {
+    if (StringUtils.isEmpty(jsFilterFn)
+        || !messageEvent.getType().equals(TopicMessageEventType.MESSAGE)) {
+      return true;
+    }
+
+    InternalTopicMessage message = messageEvent.getMessage();
+    var key = JsonNodeUtil.getJsonNodeValue(message.getKey());
+    var content = JsonNodeUtil.getJsonNodeValue(message.getContent());
+
+    try {
+      Object result = jsInvocable.invokeFunction("filter", key, content,
```

Comment:
why you choose to use function? there is an option to just bind k,v,offset,partition vars to context and invoke filtration code. It is for readability or performance?

Reply:
We can't just invoke filtration code, we have to call it as fn. I'm going to use anonymous fn to avoid collision/conflicts in parallel requests.